### PR TITLE
v3 - show historic bonus for selected location (web)

### PR DESCRIPTION
**Pull request by mathiazom**
See #612 

Renders a super simple view of the bonus data for the selected location.

## Visual Overview (Image/Video)

| Trondheim | Molde |
| --- | --- |
| image | image |

---

## Checklist

Please ensure that you’ve completed the following checkpoints before submitting your pull request:

- [x] **Documentation**: Relevant documentation has been added or updated (if applicable).
- [x] **Testing**: Have you tested your changes thoroughly?

### CODE REVIEW

File: src/compensations/Compensations.tsx [hunk 1]
```diff
@@ -64,6 +65,10 @@ const Compensations = ({ compensations, locations }: CompensationsProps) => {
     label: companyLocation.companyLocationName,
   }));
 
+  const yearlyBonusesForLocation = compensations.bonusesByLocation.find(
+    (b) => b.location._ref === selectedLocation,
+  )?.yearlyBonuses;
```
Comment (anemne) [inline src/compensations/Compensations.tsx:70]: Hva skjer om bonus yearlyBonus er undefined/null?

Comment (anemne) [reply src/compensations/Compensations.tsx:70]: Ser nå at du kun viser yearlybonus dersom det er satt for lokasjon

File: studio/lib/payloads/compensations.ts [hunk 2]
```diff
@@ -21,6 +21,13 @@ export interface SalariesPage {
   salaries: string;
 }
 
+export interface BonusesByLocationPage {
+  _type: string;
+  _key: string;
+  location: Reference;
```
Comment (anemne) [inline studio/lib/payloads/compensations.ts:27]: Jeg har skrevet det slik: location: { _ref: string; _type: string }, men det er kanskje ryddigere å bruke det som et interface?

Comment (mathiazom) [reply studio/lib/payloads/compensations.ts:27]: Ja, jeg har definert en global type ettersom dette gjelder for alle referanser.

**Review by anemne — APPROVED** [review]

**Review by christinaroise — APPROVED** [review]
Approving! 
Loving that you've used table. 👏  I'm not diving too deep into the UI part here as we know it's going to get replaced.

**Pull request merged**